src/Microsoft.Diagnostics.Monitoring.WebApi/Utilities.cs

```diff
@@ -62,6 +67,46 @@ public static KeyValueLogScope CreateArtifactScope(string artifactType, IEndpoin
             return scope;
         }
 
+        public static string GetLogsContentType(LogFormat format)
+        {
+            if (format == LogFormat.EventStream)
+            {
+                return ContentTypes.TextEventStream;
+            }
+            else if (format == LogFormat.NDJson)
+            {
+                return ContentTypes.ApplicationNdJson;
+            }
+            else if (format == LogFormat.JsonSequence)
+            {
+                return ContentTypes.ApplicationJsonSequence;
+            }
+            else
+            {
+                return ContentTypes.TextEventStream;
+            }
+        }
+
+        public static async Task StartLogsPipeline(TaskCompletionSource<object> startCompletionSource, LogFormat format, IEndpointInfo endpointInfo, EventLogsPipelineSettings settings, Stream outputStream, CancellationToken token)
+        {
+            using var loggerFactory = new LoggerFactory();
+
+            loggerFactory.AddProvider(new StreamingLoggerProvider(outputStream, format, logLevel: null));
+
+            var client = new DiagnosticsClient(endpointInfo.Endpoint);
+
+            await using EventLogsPipeline pipeline = new EventLogsPipeline(client, settings, loggerFactory);
+
+            Task pipelineTask = pipeline.RunAsync(token);
```

Comment:
Rename `pipelineTask` to `runTask` to be more descriptive of what the task represents.